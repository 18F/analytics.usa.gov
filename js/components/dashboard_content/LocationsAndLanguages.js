import React from "react";
import PropTypes from "prop-types";

import Config from "../../lib/config";
import TopCitiesRealtime from "./TopCitiesRealtime";
import TopCountriesRealtime from "./TopCountriesRealtime";
import TopLanguagesHistorical from "./TopLanguagesHistorical";

/**
 * Contains charts and other data visualizations for the user locations and
 * languages section of the site. This component is mainly laying out the
 * structure for the section and passes props necessary for getting data and
 * displaying visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function LocationsAndLanguages({ dataHrefBase }) {
  return (
    <>
      <div className="section__headline">
        <h2>User Locations and Languages in the Last 30 Minutes</h2>
      </div>

      <div className="section__chart grid-row">
        <section
          id="cities"
          className="desktop:grid-col-4 padding-2 bar-chart-component"
        >
          <h4>Cities</h4>
          <TopCitiesRealtime
            dataHrefBase={dataHrefBase}
            refreshSeconds={Config.realtimeDataRefreshSeconds}
          />
        </section>

        <section
          id="countries"
          className="desktop:grid-col-4 padding-2 bar-chart-component"
        >
          <h4>Countries</h4>
          <TopCountriesRealtime
            dataHrefBase={dataHrefBase}
            refreshSeconds={Config.realtimeDataRefreshSeconds}
          />
        </section>

        <section
          id="languages"
          className="desktop:grid-col-4 padding-2 bar-chart-component"
        >
          <h4>Languages</h4>
          <TopLanguagesHistorical dataHrefBase={dataHrefBase} />
        </section>
      </div>
    </>
  );
}

LocationsAndLanguages.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default LocationsAndLanguages;
