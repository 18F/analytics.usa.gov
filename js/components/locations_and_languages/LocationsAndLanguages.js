import React from "react";
import PropTypes from "prop-types";

import TopCitiesRealtime from "./TopCitiesRealtime";
import TopCountriesRealtime from "./TopCountriesRealtime";
import TopLanguagesHistorical from "./TopLanguagesHistorical";

function LocationsAndLanguages({ dataURL, dataPrefix }) {
  const dataHrefBase = `${dataURL}/${dataPrefix}`;

  return (
    <>
      <div className="section__headline">
        <h2>User Locations and Languages in the Last 30 Minutes</h2>
      </div>

      <div className="section__chart grid-row">
        <section
          id="cities"
          className="desktop:grid-col-4 padding-2 bar-chart-component"
        >
          <h4>Cities</h4>
          <TopCitiesRealtime dataHrefBase={dataHrefBase} />
        </section>

        <section
          id="countries"
          className="desktop:grid-col-4 padding-2 bar-chart-component"
        >
          <h4>Countries</h4>
          <TopCountriesRealtime dataHrefBase={dataHrefBase} />
        </section>

        <section
          id="languages"
          className="desktop:grid-col-4 padding-2 bar-chart-component"
        >
          <h4>Languages</h4>
          <TopLanguagesHistorical dataHrefBase={dataHrefBase} />
        </section>
      </div>
    </>
  );
}

LocationsAndLanguages.propTypes = {
  dataURL: PropTypes.string.isRequired,
  dataPrefix: PropTypes.string.isRequired,
};

export default LocationsAndLanguages;
