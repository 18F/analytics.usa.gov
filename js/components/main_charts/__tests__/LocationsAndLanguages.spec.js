import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import LocationsAndLanguages from "../LocationsAndLanguages";

describe("LocationsAndLanguages", () => {
  let component;

  beforeEach(async () => {
    component = render(
      <LocationsAndLanguages dataHrefBase="http://www.example.com/data/" />,
    );
    await waitFor(() => screen.getByText("Languages"));
  });

  it("renders", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
