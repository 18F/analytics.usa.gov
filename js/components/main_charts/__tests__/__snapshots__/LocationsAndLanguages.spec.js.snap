// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsAndLanguages renders 1`] = `
<DocumentFragment>
  <div
    class="section__headline"
  >
    <h2>
      User Locations and Languages in the Last 30 Minutes
    </h2>
  </div>
  <div
    class="section__chart grid-row"
  >
    <section
      class="desktop:grid-col-4 padding-2 bar-chart-component"
      id="cities"
    >
      <h4>
        Cities
      </h4>
      <figure
        id="chart_top-cities-realtime"
      >
        <div
          class="data bar-chart"
        />
      </figure>
    </section>
    <section
      class="desktop:grid-col-4 padding-2 bar-chart-component"
      id="countries"
    >
      <h4>
        Countries
      </h4>
      <div>
        <figure
          id="chart_us"
        >
          <div
            class="data bar-chart"
          />
        </figure>
        <figure
          class="hide"
          id="chart_us_and_territories"
        >
          <h4>
            United States & Territories
          </h4>
          <div
            class="data bar-chart"
          />
        </figure>
        <figure
          class="hide"
          id="chart_countries"
        >
          <h4>
            International
          </h4>
          <div
            class="data bar-chart"
          />
        </figure>
      </div>
    </section>
    <section
      class="desktop:grid-col-4 padding-2 bar-chart-component"
      id="languages"
    >
      <h4>
        Languages
      </h4>
      <figure
        data-block="languages"
        id="chart_top-languages"
      >
        <div
          class="data bar-chart"
        />
      </figure>
    </section>
  </div>
</DocumentFragment>
`;
